benchmark/infrastructure/BenchmarkConfiguration.h

```diff
@@ -5,28 +5,135 @@
+        return key >= 0;
+      } else {
+        return true;
+      }
+    };
+
+    return (biggerOrEqual(keys) && ...);
+  }
+}
+
 /*
- * A rather basic wrapper for nlohmann::json, which only allows reading of
- * information and setting the configuration by parsing strings.
- */
+Manages a bunch of `BenchmarkConfigurationOption`s.
+*/
 class BenchmarkConfiguration {
-  // No real reason, to really build everything ourselves, when the
-  // nlohmann::json object already containes everything, that we could need.
-  // P.S.: It initializes to an empty json object.
-  nlohmann::json data_ = nlohmann::json(nlohmann::json::value_t::object);
+  // The added configuration options.
+  std::vector<BenchmarkConfigurationOption> configurationOptions_;
+
+  /*
+  A configuration option tends to be placed like a key value pair in a json
+  object. For example: `{"object 1" : [{"object 2" : { "The configuration option
+  identifier" : "Configuration information"} }]}`
+
+  That is, only accesable through the usage of multiple keys in the form of
+  strings and numbers. Like in a container modeling a tree, with the
+  configuration option in the leaves.
+
+  This `nlohmann::json` object describes their `position` inside those trees, by
+  having their `path` end in a key value pair, with the key being the
+  configuration options identifier and the value their index inside
+  `configurationOptions_`.
+  */
+  nlohmann::json keyToConfigurationOptionIndex_ =
+      nlohmann::json(nlohmann::json::value_t::object);
+
+  /*
+  @brief Creates a valid `nlohmann::json` pointer based on  the given keys.
+
+  @tparam Keys Positive whole numbers, or strings.
+  */
+  static nlohmann::json::json_pointer createJsonPointer(
+      const KeyForJson auto&... keys) requires(sizeof...(keys) > 0) {
+    // A numeric key, must be `>=0`.
+    AD_CONTRACT_CHECK(allArgumentsBiggerOrEqualToZero(keys...));
+
+    /*
+    A json pointer needs special characters, if a `/`, or `~`, is used in a
+    key. So here a special conversion function for our keys, that adds those,
+    if needed.
+    */
+    auto toString = []<typename T>(const T& key) {
+      // Our transformed key.
+      std::string transformedKey;
+
+      /*
+      Transforming the key. We simply check through the way, we can convert
+      them into a string and do the one, that works first.
+      */
+      if constexpr (std::is_constructible_v<std::string, T>) {
+        transformedKey = std::string(key);
+      } else {
+        /*
+        Must have been a number. I mean, `KeyForJson` doesn't allow anything
+        else, than those 3 possibilities and this the the last one.
+        */
+        transformedKey = std::to_string(key);
+      }
+
+      // Replace special character `~` with `~0`.
+      transformedKey = absl::StrReplaceAll(transformedKey, {{"~", "~0"}});
+
+      // Replace special character `/` with `~1`.
+      return absl::StrReplaceAll(transformedKey, {{"/", "~1"}});
```

Comment:
Can be a single call to `StrReplaceAll` instead of two.

Reply:
Done, see [3a313c4](https://github.com/ad-freiburg/qlever/pull/977/commits/3a313c4f34bd9be95f3e8636e88e0bf7d952cfa7).